src/main/java/org/jabref/gui/filelist/AttachFileAction.java

```diff
@@ -27,9 +24,14 @@ public void action() {
         }
         BibEntry entry = panel.getSelectedEntries().get(0);
         LinkedFile flEntry = new LinkedFile("", "", "");
-        FileListEntryEditor editor = new FileListEntryEditor(flEntry, false, true,
+
+        FileListDialogView dlg = new FileListDialogView();
+
+        Platform.runLater(() -> dlg.show());
+        /*   FileListEntryEditor editor = new FileListEntryEditor(flEntry, false, true,
                 panel.getBibDatabaseContext());
         editor.setVisible(true, true);
+
         if (editor.okPressed()) {
```

Comment:
@tobiasdiez  How do I handle such cases in javafx?  The check for a return value? Should I do a showAndWait?
 My initial idea would be to pass the UndoManager to the javafx view model (state manager? or other injection)
There are some other clases in which a similar stuff is executed which seems to a bit more complicated, e.g. the
DownloadExternalFile class

Reply:
Good question. I think the return value should be accessed per getters of the `FileListDialogView` (this seems to be the most flexible and easiest solution - you don't need to create "Dialog return value classes" for more complex scenarios). These getters in the view should ask the view model of the dialog about the values. You can get the controller with 
https://github.com/JabRef/jabref/blob/1172ed493c6dce05fcb427b4e665f5de3a7b3fc3/src/main/java/org/jabref/gui/AbstractView.java#L45
(just change the visibility) and then the view model with
https://github.com/JabRef/jabref/blob/1172ed493c6dce05fcb427b4e665f5de3a7b3fc3/src/main/java/org/jabref/gui/AbstractController.java#L20

For parameter injection, have a look at `CopyFilesDialogView`.